main.go

```diff
@@ -97,7 +97,12 @@ func main() {
 	}()
 
 	t0 := time.Now() // before dns resolution
-	raddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", host, port))
+
```

Comment:
I'm sorry this fix is not correct. You want to fix the logic inside the function at 87 to split hostport at that point.

There are three cases to deal with

`host`, then return host and choose the port based on the scheme, this is effectively what the code does now

`host:port`, split on the colon, return the first part as the  host, and strconv.Atoi the second part to be the port. There may be a helper function in the net or net/http package that does this.

`:port`: it's an error.
`host:` is also an error